src/core/random.js

```diff
@@ -0,0 +1,105 @@
+import seedrandom from 'seedrandom';
+
+function getrandomfn(ctx) {
+  let randomfn;
+  if (ctx.prngstate === undefined) {
+    // no call to a random function has been made.
+    // pre-populate the state info
+    // TODO what if ctx.seed is not defined?
+    randomfn = new seedrandom.alea(ctx.seed, { state: true });
+  } else {
+    randomfn = new seedrandom.alea('', { state: ctx.prngstate });
+  }
+  return randomfn;
+}
+
+/**
+ * random
+ *
+ * Generates a random number.
+ * The seed and current state is kept inside the ctx object.
+ *
+ * @param {...object} ctx - The context object keeping the seed and current state.
+ */
+function random(ctx) {
+  const r = getrandomfn(ctx);
+  const randomnumber = r();
+  const ctx2 = { ...ctx, prngstate: r.state() };
+  return { randomnumber, ctx: ctx2 };
+}
+
+/**
+ * shuffle
+ *
+ * Shuffles a given array.
+ * The seed and current state is kept inside the ctx object.
+ *
+ * Please note this is an in-place shuffle. It will modify the given array.
+ * If you want to retain the original order, clone the array before.
+ *
+ * This code is taken directly from Mike Bostock, with a few slight modifications
+ * https://bost.ocks.org/mike/shuffle/
+ *
+ * @param {...object} ctx - The context object keeping the seed and current state.
+ * @param {any[]} array - The array to sort. Will be modified in-place.
+ */
+function shuffle(ctx, array) {
```

Comment:
If this is meant to be a public API, then it probably should look similar to `rolldie` below (i.e. only place a request to shuffle in `G`).

Reply:
I've currently dropped shuffle for now, since this is something that could be done via `random` + a kind-of _seeded_ shuffle implementation.